src/test/java/pl/zaradny/springApp/api/ProductEndpointTest.java

```diff
@@ -0,0 +1,160 @@
+import pl.zaradny.springApp.domain.ProductResponseDto;
+import pl.zaradny.springApp.exceptions.ProductNotFoundException;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class ProductEndpointTest extends SpringAppApplicationTests {
+
+
+    @Autowired
+    ProductFacade productFacade;
+
+    @Test
+    public void shouldGetExistingProduct(){
+        //given
+        ProductRequestDto requestDto = new ProductRequestDto("product");
+        ProductResponseDto existingProduct = productFacade.create(requestDto);
+        final String url = "http://localhost:"+ port + "/products/" + existingProduct.getId();
+
+        //when
+        ResponseEntity<ProductResponseDto> result = httpClient.getForEntity(url, ProductResponseDto.class);
+
+        //then
+        assertThat(result.getStatusCodeValue()).isEqualTo(200);
+        assertThat(result.getBody()).isEqualToComparingFieldByField(existingProduct);
+    }
+
+    @Test
+    public void shouldResponse404HttpCodeWhenProductDoesNotExist(){
+        //given
+        ProductRequestDto requestDto = new ProductRequestDto("product");
+        ProductResponseDto existingProduct = productFacade.create(requestDto);
+        productFacade.deleteById(existingProduct.getId());
+        final String url = "http://localhost:"+ port + "/products/" + existingProduct.getId();
+
+        //when
+        ResponseEntity<ProductResponseDto> result = httpClient.getForEntity(url, ProductResponseDto.class);
+
+        //then
+        assertThat(result.getStatusCodeValue()).isEqualTo(404);
+    }
+
+    @Test
+    public void shouldCreateProduct(){
+        //given
+        final String url = "http://localhost:"+ port + "/products";
+        final ProductRequestDto product = new ProductRequestDto("iphone");
+        String productJson = mapToJson(product);
+
+        //when
+        ResponseEntity<ProductResponseDto> result = httpClient.postForEntity(url, getHttpRequest(productJson), ProductResponseDto.class);
+
+        //then
+        assertThat(result.getStatusCodeValue()).isEqualTo(200);
+        assertThat(result.getBody().getName()).isEqualTo("iphone");
+    }
+
+    @Test(expected = ProductNotFoundException.class)
+    public void shouldDeleteExistingProduct(){
+        //given
+        ProductRequestDto requestDto = new ProductRequestDto("product");
+        ProductResponseDto existingProduct = productFacade.create(requestDto);
+        final String url = "http://localhost:"+ port + "/products/" + existingProduct.getId();
+
+        //when
+        ResponseEntity<ProductResponseDto> result = httpClient.exchange(url,
+                HttpMethod.DELETE,
+                HttpEntity.EMPTY,
+                ProductResponseDto.class );
+
+        //then
+        assertThat(result.getStatusCodeValue()).isEqualTo(200);
+        assertThat(result.getBody()).isEqualToComparingFieldByField(existingProduct);
+        productFacade.findById(existingProduct.getId());
```

Comment:
Co to robi? :)

Reply:
Po dodaniu JUnit5 test jest już czytelniejszy :)